src/field255.rs

```diff
@@ -0,0 +1,308 @@
+
+        let mut self_encoded = [0; 32];
+        fiat_25519_to_bytes(&mut self_encoded, &self.0);
+        self_encoded.hash(state);
+    }
+}
+
+impl Add for Field255 {
+    type Output = Field255;
+
+    fn add(self, rhs: Self) -> Field255 {
+        let mut temp = [0; 5];
+        fiat_25519_add(&mut temp, &self.0, &rhs.0);
+        let mut output = [0; 5];
+        fiat_25519_carry(&mut output, &temp);
+        Field255(output)
+    }
+}
+
+impl AddAssign for Field255 {
+    fn add_assign(&mut self, rhs: Self) {
+        let mut temp = [0; 5];
+        fiat_25519_add(&mut temp, &self.0, &rhs.0);
+        fiat_25519_carry(&mut self.0, &temp);
+    }
+}
+
+impl Sub for Field255 {
+    type Output = Field255;
+
+    fn sub(self, rhs: Self) -> Field255 {
+        let mut temp = [0; 5];
+        fiat_25519_sub(&mut temp, &self.0, &rhs.0);
+        let mut output = [0; 5];
+        fiat_25519_carry(&mut output, &temp);
+        Field255(output)
+    }
+}
+
+impl SubAssign for Field255 {
+    fn sub_assign(&mut self, rhs: Self) {
+        let mut temp = [0; 5];
+        fiat_25519_sub(&mut temp, &self.0, &rhs.0);
+        fiat_25519_carry(&mut self.0, &temp);
+    }
+}
+
+impl Mul for Field255 {
+    type Output = Field255;
+
+    fn mul(self, rhs: Self) -> Field255 {
+        let mut output = [0; 5];
+        fiat_25519_carry_mul(&mut output, &self.0, &rhs.0);
+        Field255(output)
+    }
+}
+
+impl MulAssign for Field255 {
+    fn mul_assign(&mut self, rhs: Self) {
+        let self_copy = self.0;
+        fiat_25519_carry_mul(&mut self.0, &self_copy, &rhs.0);
+    }
+}
+
+impl Div for Field255 {
+    type Output = Field255;
+
+    fn div(self, _rhs: Self) -> Self::Output {
+        unimplemented!()
+    }
+}
+
+impl DivAssign for Field255 {
+    fn div_assign(&mut self, _rhs: Self) {
+        unimplemented!()
+    }
+}
+
+impl Neg for Field255 {
+    type Output = Field255;
+
+    fn neg(self) -> Field255 {
+        let mut temp = [0; 5];
+        fiat_25519_opp(&mut temp, &self.0);
+        let mut output = [0; 5];
+        fiat_25519_carry(&mut output, &temp);
+        Field255(output)
+    }
+}
+
+impl<'a> TryFrom<&'a [u8]> for Field255 {
+    type Error = FieldError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, FieldError> {
+        Self::try_from_bytes(bytes, false)
+    }
+}
+
+impl From<Field255> for [u8; Field255::ENCODED_SIZE] {
+    fn from(element: Field255) -> Self {
+        let mut array = [0; 32];
+        fiat_25519_to_bytes(&mut array, &element.0);
+        array.reverse();
+        array
+    }
+}
+
+impl From<Field255> for Vec<u8> {
+    fn from(elem: Field255) -> Vec<u8> {
+        <[u8; Field255::ENCODED_SIZE]>::from(elem).to_vec()
+    }
+}
+
+impl Display for Field255 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let bytes: [u8; 32] = (*self).into();
+        let bigint = BigInt::from_bytes_be(Sign::Plus, bytes.as_slice());
```

Comment:
Instead of converting to decimal, WDYTA simply printing the big-endian byte string, or even converting to hex? It would be nice to avoid the dependency on `BigInt` (except in tests of course, which you suggested would be needed.)